astropy/convolution/convolve.py

```diff
@@ -665,25 +684,24 @@ def convolve_fft(array, kernel, boundary='fill', fill_value=0.,
         raise NotImplementedError("The 'extend' option is not implemented "
                                   "for fft-based convolution")
 
-    # find ideal size (power of 2) for fft.
-    # Can add shapes because they are tuples
+    # Find ideal size for fft (was power of 2, now any powers of prime factors 2, 3, 5).
+    # Add shapes elementwise for psf_pad.
     if fft_pad:  # default=True
         if psf_pad:  # default=False
-            # add the dimensions and then take the max (bigger)
-            fsize = 2 ** np.ceil(np.log2(
-                np.max(np.array(arrayshape) + np.array(kernshape))))
+            # add the sizes along each dimension (bigger)
+            fsize = np.array(arrayshape) + np.array(kernshape)
         else:
-            # add the shape lists (max of a list of length 4) (smaller)
-            # also makes the shapes square
-            fsize = 2 ** np.ceil(np.log2(np.max(arrayshape + kernshape)))
-        newshape = np.full((array.ndim, ), fsize, dtype=int)
+            # take the larger shape in each dimension (smaller)
+            fsize = np.maximum(arrayshape, kernshape)
+        # newshape = np.array([2 ** np.ceil(np.log2(i)).astype(int) for i in fsize])
+        # Get optimized sizes from scipy.
+        newshape = _next_fast_lengths(fsize)
     else:
         if psf_pad:
             # just add the biggest dimensions
             newshape = np.array(arrayshape) + np.array(kernshape)
         else:
-            newshape = np.array([np.max([imsh, kernsh])
-                                 for imsh, kernsh in zip(arrayshape, kernshape)])
+            newshape = np.maximum(arrayshape, kernshape)
```

Comment:
`np.maximum` is new to me, that's a great usage here.

This is functionally equivalent, right?

Reply:
yes